### PR TITLE
Added code to implement rename functionality for map operator

### PR DESCRIPTION
**Pull request by chiragsahni**

### CODE REVIEW

File: lale/lib/lale/map.py [hunk 1]
```diff
@@ -29,6 +29,8 @@ def transform(self, X):
         def get_map_function_output(column, new_column_name):
             functions_module = importlib.import_module("lale.lib.lale.functions")
             function_name = column._expr.func.id
+            if _is_ast_attribute(column):
```
Comment (shinnar) [inline lale/lib/lale/map.py:32]: is the line to support `columns={"new": it.old}` (without an explicit rename?)  if so, that makes sense, but please add a test for it

Comment (shinnar) [inline lale/lib/lale/map.py:32]: also, what is the expected behavior of 
`columns=[rename(it.old)]`
`columns=[it.old]`.
We may make sure they do something reasonable and add a test for them.

**Review by shinnar — COMMENTED** [review]
in general, looks good.  A few questions in line

File: lale/lib/lale/map.py (hunk 1 above)
Comment (chiragsahni) [reply lale/lib/lale/map.py:32]: Na.. Actually left it by mistake. I will remove it as we want only explicit rename.

**Review by shinnar — APPROVED** [review]
changes look good

**Review by shinnar — COMMENTED** [review]
please include a test with an explicit call to the identity function

**Pull request merged**